homeassistant/components/binary_sensor/apcupsd.py

```diff
@@ -0,0 +1,46 @@
+"""
+homeassistant.components.binary_sensor.apcupsd
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Provides a binary sensor to track online status of a UPS.
+"""
+from homeassistant.core import JobPriority
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components import apcupsd
+
+
+DEPENDENCIES = [apcupsd.DOMAIN]
+
+DEFAULT_NAME = "UPS Online Status"
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """ Instantiate an OnlineStatus binary sensor entity and add it to HA. """
+    add_entities((OnlineStatus(hass, config),))
+
+
+class OnlineStatus(BinarySensorDevice):
+    """ Binary sensor to represent UPS online status. """
+    def __init__(self, hass, config):
+        self._config = config
+        self._state = None
+        # Get initial state
+        hass.pool.add_job(
```

Comment:
No need for this. update state is always called after an entity gets added: https://github.com/balloob/home-assistant/blob/dev/homeassistant/helpers/entity_component.py#L106-L121

Just make sure you call `self.update()` here.